src/Package.php

```diff
@@ -346,6 +414,60 @@ public function boot(Module ...$defaultModules): bool
         return true;
     }
 
+    /**
+     * @param Module ...$defaultModules
+     * @return void
+     */
+    private function doBuild(Module ...$defaultModules): void
+    {
+        if ($defaultModules) {
+            $this->deprecatedArgument(
+                sprintf(
+                    'Passing default modules to %1$s::boot() is deprecated since version 1.7.0.'
+                    . ' Please add modules via %1$s::addModule().',
+                    __CLASS__
+                ),
+                __METHOD__,
+                '1.7.0'
+            );
+        }
+
+        if (!$this->built) {
+            array_map([$this, 'addModule'], $defaultModules);
+            $this->build();
+
+            return;
+        }
+
+        if (
+            !$defaultModules
+            || ($this->status >= self::STATUS_MODULES_ADDED)
+            || ($this->status === self::STATUS_FAILED)
```

Comment:
The call `$this->status === self::STATUS_FAILED` can be replaced by `$this->statusIs(self::STATUS_FAILED)`. I'm wondering if we can extend this method by `statusIs(string $status, string $operator = '===')` and use it with `version_compare()` like `assertStatus()` 🤔

Reply:
@Chrico regarding

> $this->status === self::STATUS_FAILED can be replaced by $this->statusIs(self::STATUS_FAILED). 

I can change that.

Reply:
Totally fine as it is. Not really a blocker, just an "improvement". ✔️